Makefile

```diff
@@ -122,7 +122,14 @@ TEXT_ENGINE := none
 ifeq ($(TEXT_ENGINE), s2dex_text_engine)
   DEFINES += S2DEX_GBI_2=1 S2DEX_TEXT_ENGINE=1
   LIBRARIES += s2d_engine
-  DUMMY != make -C src/s2d_engine COPY_DIR=$(shell pwd)/lib/
+  # less hacky way to do this?
+  LIBS_DIR += -L src/s2d_engine/build
+  DUMMY != ls src/s2d_engine/init.c >&2 || echo FAIL
+    ifeq ($(DUMMY),FAIL)
+      $(info Grabbing S2DEX Text Engine from `https://github.com/someone2639/S2DEX-Text-Engine`......)
+      D2 != $(shell git clone -b HackerSM64 https://github.com/someone2639/S2DEX-Text-Engine src/s2d_engine)
+    endif
+  DUMMY != make -C src/s2d_engine COPY_DIR=$(shell pwd)/$(BUILD_DIR)/lib/
```

Comment:
Building the text engine files using this dummy command means 1) they can't be parallelized alongside the rest of the repo and 2) they'll build no matter what target is passed, including clean. The better solution is to make a rule for the archive that the s2d engine builds, just like libgoddard has, and then run `make -C src/s2d_engine` in the rule for that archive. That rule can then have the s2d engine folder as a `|` dependency, and the s2d engine folder can have a rule to clone the s2d engine repo. This would let you get rid of both DUMMY commands.